scipy/stats/morestats.py

```diff
@@ -2074,17 +2074,88 @@ def anderson_ksamp(samples, midrank=True):
+            self.n, self.m = n, m
+            # distribution is NOT symmetric when m + n is odd
+            # n is len(x), m is len(y), and ratio of scales is defined x/y
+            astart, a1, _ = statlib.gscale(n, m)
+            self.astart = astart  # minimum value of statistic
+            # Exact distribution of test statistic under null hypothesis
+            # expressed as frequencies/counts/integers to maintain precision.
+            # Stored as floats to avoid overflow of sums.
+            self.freqs = a1.astype(np.float64)
+            self.total = self.freqs.sum()  # could calculate from m and n
+            # probability mass is self.freqs / self.total;
+
+    def pmf(self, k, n, m):
+        """Probability mass function."""
+        self._recalc(n, m)
+        # The convention here is that PMF at k = 12.5 is the same as at k = 12,
+        # -> use `floor` in case of ties.
+        ind = np.floor(k - self.astart).astype(int)
+        return self.freqs[ind] / self.total
+
+    def cdf(self, k, n, m):
+        """Cumulative distribution function."""
+        self._recalc(n, m)
+        # This is a bit subtle; easiest to think through an example:
+        # the CDF at k = 12.5 should be the same as at k = 12 -> `floor`
+        # The convention is for CDF at k = 12 to included the probability mass
+        # at k = 12 -> slice `:ind+1`.
+        ind = np.floor(k - self.astart).astype(int)
+        return self.freqs[:ind+1].sum() / self.total
+
+    def sf(self, k, n, m):
+        """Survival function."""
+        self._recalc(n, m)
+        # This is a bit subtle; easiest to think through an example:
+        # the SF at k = 12.5 should be the same at k = 13 -> `ceil`.
+        # The convention is for  SF at k = 13 to included the probability mass
+        # at k = 13 -> slice `ind:`.
+        ind = np.ceil(k - self.astart).astype(int)
+        return self.freqs[ind:].sum() / self.total
+
+
+# Maintain state for faster repeat calls to ansari w/ method='exact'
+_abw_state = _ABW()
+
+
+def ansari(x, y, alternative='two-sided'):
     """Perform the Ansari-Bradley test for equal scale parameters.
 
     The Ansari-Bradley test ([1]_, [2]_) is a non-parametric test
     for the equality of the scale parameter of the distributions
-    from which two samples were drawn.
+    from which two samples were drawn. The null hypothesis states that
+    the ratio of the scale of the distribution underlying `x` to the scale
+    of the distribution underlying `y` is 1.
 
     Parameters
     ----------
     x, y : array_like
         Arrays of sample data.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis. Default is 'two-sided'.
+        The following options are available:
+
+        * 'two-sided': the ratio of scales is not equal to 1.
+        * 'less': the ratio of scales is less than 1.
+        * 'greater': the ratio of scales is greater than 1.
```

Comment:
I am seeing this in another PR. Not sure though it's needed.
```suggestion
        * 'greater': the ratio of scales is greater than 1.

        .. versionadded:: 1.7.0
```

Reply:
This would be nice to have this. It adds a nice label to tell the users from which version onwards the functionality is available. 


![2021-04-15_15 40 13](https://user-images.githubusercontent.com/43181252/114853193-2e11b380-9e01-11eb-8110-a6a523f8be88.png)

Reply:
Admittedly we're not consistent about adding it thought.